test/unit/specs/components/BodySystemForm.spec.js

```diff
@@ -0,0 +1,84 @@
+import Vuex from 'vuex';
+import {createLocalVue, shallow} from '@vue/test-utils';
+
+import BodySystemForm from '@/components/BodySystemForm';
+import bodySystems from '@/store/systems';
+
+const localVue = createLocalVue();
+localVue.use(Vuex);
+
+describe('BodySystemForm.vue', () => {
+  let store, state, mutations;
+
+  beforeEach(() => {
+    state = {
+      selectedSystems: bodySystems.slice(1)
+    };
+
+    mutations = {
+      toggleSystem: jest.fn()
+    };
+
+    store = new Vuex.Store({state, mutations});
+  });
+
+  describe('rendering', () => {
+    test('renders all systems as checkboxes', () =>{
+      const wrapper = shallow(BodySystemForm, {localVue, store});
+      expect(wrapper.findAll('input[type=checkbox]').length).toBe(bodySystems.length);
+    });
+
+    test('submitButton is disabled', () => {
+      const wrapper = shallow(BodySystemForm, {localVue, store});
+      const submitButton = wrapper.find('input[type=button]');
+      expect(submitButton.attributes().disabled).toBeUndefined();
+
+      state.selectedSystems = [];
+      expect(submitButton.attributes().disabled).toBeDefined();
+    });
+
+    // TODO if the system is in the store it is checked
```

Comment:
I got tripped over interacting with the state over the checkboxes.

Reply:
You should be able to just mutate `state.selectedSystems` as done in the test above, then use `wrapper.findAll` to count the number of checkboxes that are checked. If you add a commented test, I could probably tell you where you're going wrong.

Reply:
The following works, but due to the persnickety rules about what constitutes a valid CSS selector, it might be simpler if the logic were flipped. That is, find all the checked checkboxes, then check that the ID is in the array of body systems. But whatever.

```javascript
test('checkboxes for selected systems are checked', () => {
  const selectedSystems = bodySystems.slice(0, 3);
  state.selectedSystems = selectedSystems;
  const wrapper = shallow(BodySystemForm, { localVue, store });
  selectedSystems.forEach(system => {
    const inputSelector = `input#${system.id.replace(':', '\\:')}`;
    const checkbox = wrapper.find(inputSelector);
    expect(checkbox.attributes('checked')).toBeTruthy();
  });
});
```

Reply:
You probably would also want to verify that exactly three checkboxes are checked. You get the idea.